sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/ApproximatePercentile.scala

```diff
@@ -83,32 +83,32 @@ case class ApproximatePercentile(
   }
 
   // Mark as lazy so that accuracyExpression is not evaluated during tree transformation.
-  private lazy val accuracy: Int = accuracyExpression.eval().asInstanceOf[Int]
+  private lazy val accuracy: Long = accuracyExpression.eval().asInstanceOf[Long]
```

Comment:
what about other systems? Does `accuracy` really need to be long type?

Reply:
I checked some, Hive use integrals, presto use decimals, oracle, snowflake do not expose accuracy.

Reply:
How about we only accept up to int type? This was the expectation before, and it's foldable, so we can allow in in type coercion but fail in `checkInputDataTypes` if it exceeds Int.Max

Reply:
I have considered this. Hive accects 1L , 2L, etc, although when specify a large accuracy number(no need to Int.MaxValue) will cause failure(maybe oom). I'm ok with just int.

Reply:
```sql
0: jdbc:hive2://quickstart.cloudera:10000> SELECT percentile_approx(10.0, array(0.9,0.99), 2147483647L);
INFO  : Compiling command(queryId=hive_20191217143232_9c12b057-9d23-4a8d-bd00-0954a0c23b19): SELECT percentile_approx(10.0, array(0.9,0.99), 2147483647L)
INFO  : Semantic Analysis Completed
INFO  : Returning Hive schema: Schema(fieldSchemas:[FieldSchema(name:_c0, type:array<double>, comment:null)], properties:null)
INFO  : Completed compiling command(queryId=hive_20191217143232_9c12b057-9d23-4a8d-bd00-0954a0c23b19); Time taken: 0.052 seconds
INFO  : Concurrency mode is disabled, not creating a lock manager
INFO  : Executing command(queryId=hive_20191217143232_9c12b057-9d23-4a8d-bd00-0954a0c23b19): SELECT percentile_approx(10.0, array(0.9,0.99), 2147483647L)
INFO  : Query ID = hive_20191217143232_9c12b057-9d23-4a8d-bd00-0954a0c23b19
INFO  : Total jobs = 1
INFO  : Launching Job 1 out of 1
INFO  : Starting task [Stage-1:MAPRED] in serial mode
INFO  : Number of reduce tasks determined at compile time: 1
INFO  : In order to change the average load for a reducer (in bytes):
INFO  :   set hive.exec.reducers.bytes.per.reducer=<number>
INFO  : In order to limit the maximum number of reducers:
INFO  :   set hive.exec.reducers.max=<number>
INFO  : In order to set a constant number of reducers:
INFO  :   set mapreduce.job.reduces=<number>
INFO  : Starting Job = job_1576485475891_0016, Tracking URL = http://quickstart.cloudera:8088/proxy/application_1576485475891_0016/
INFO  : Kill Command = /usr/lib/hadoop/bin/hadoop job  -kill job_1576485475891_0016
INFO  : Hadoop job information for Stage-1: number of mappers: 1; number of reducers: 1
INFO  : 2019-12-17 14:32:41,438 Stage-1 map = 0%,  reduce = 0%
INFO  : 2019-12-17 14:32:47,608 Stage-1 map = 100%,  reduce = 0%, Cumulative CPU 1.92 sec
INFO  : 2019-12-17 14:32:53,786 Stage-1 map = 100%,  reduce = 100%, Cumulative CPU 3.43 sec
INFO  : MapReduce Total cumulative CPU time: 3 seconds 430 msec
INFO  : Ended Job = job_1576485475891_0016
INFO  : MapReduce Jobs Launched:
INFO  : Stage-Stage-1: Map: 1  Reduce: 1   Cumulative CPU: 3.43 sec   HDFS Read: 7504 HDFS Write: 10 SUCCESS
INFO  : Total MapReduce CPU Time Spent: 3 seconds 430 msec
INFO  : Completed executing command(queryId=hive_20191217143232_9c12b057-9d23-4a8d-bd00-0954a0c23b19); Time taken: 19.88 seconds
INFO  : OK
+--------------+--+
|     _c0      |
+--------------+--+
| [10.0,10.0]  |
+--------------+--+
1 row selected (19.957 seconds)
0: jdbc:hive2://quickstart.cloudera:10000> SELECT percentile_approx(10.0, array(0.9,0.99), 2147483648L);
INFO  : Compiling command(queryId=hive_20191217143333_83ece458-15af-47a6-85f4-2299b9cb170b): SELECT percentile_approx(10.0, array(0.9,0.99), 2147483648L)
INFO  : Semantic Analysis Completed
INFO  : Returning Hive schema: Schema(fieldSchemas:[FieldSchema(name:_c0, type:array<double>, comment:null)], properties:null)
INFO  : Completed compiling command(queryId=hive_20191217143333_83ece458-15af-47a6-85f4-2299b9cb170b); Time taken: 0.042 seconds
INFO  : Concurrency mode is disabled, not creating a lock manager
INFO  : Executing command(queryId=hive_20191217143333_83ece458-15af-47a6-85f4-2299b9cb170b): SELECT percentile_approx(10.0, array(0.9,0.99), 2147483648L)
INFO  : Query ID = hive_20191217143333_83ece458-15af-47a6-85f4-2299b9cb170b
INFO  : Total jobs = 1
INFO  : Launching Job 1 out of 1
INFO  : Starting task [Stage-1:MAPRED] in serial mode
INFO  : Number of reduce tasks determined at compile time: 1
INFO  : In order to change the average load for a reducer (in bytes):
INFO  :   set hive.exec.reducers.bytes.per.reducer=<number>
INFO  : In order to limit the maximum number of reducers:
INFO  :   set hive.exec.reducers.max=<number>
INFO  : In order to set a constant number of reducers:
INFO  :   set mapreduce.job.reduces=<number>
INFO  : Starting Job = job_1576485475891_0017, Tracking URL = http://quickstart.cloudera:8088/proxy/application_1576485475891_0017/
INFO  : Kill Command = /usr/lib/hadoop/bin/hadoop job  -kill job_1576485475891_0017
INFO  : Hadoop job information for Stage-1: number of mappers: 1; number of reducers: 1
INFO  : 2019-12-17 14:33:08,805 Stage-1 map = 0%,  reduce = 0%
INFO  : 2019-12-17 14:33:31,372 Stage-1 map = 100%,  reduce = 100%
ERROR : Ended Job = job_1576485475891_0017 with errors
ERROR : FAILED: Execution Error, return code 2 from org.apache.hadoop.hive.ql.exec.mr.MapRedTask
INFO  : MapReduce Jobs Launched:
INFO  : Stage-Stage-1: Map: 1  Reduce: 1   HDFS Read: 0 HDFS Write: 0 FAIL
INFO  : Total MapReduce CPU Time Spent: 0 msec
INFO  : Completed executing command(queryId=hive_20191217143333_83ece458-15af-47a6-85f4-2299b9cb170b); Time taken: 30.112 seconds
Error: Error while processing statement: FAILED: Execution Error, return code 2 from org.apache.hadoop.hive.ql.exec.mr.MapRedTask (state=08S01,code=2)
```
rechecked with hive, value greater than Int.MaxValue is invalid